Homework5/SparseMatrix/SparseMatrix.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SMatrix
+{
+    internal class SparseMatrix : IEnumerable<int>
+    {
+
+        public int RowIndex { get; private set; }
+        public int ColumnIndex { get; private set; }
+
+        public List<(int row, int column, int value)> Elements { get; set; }
+
+        public SparseMatrix(int rowIndex, int colIndex)
+        {
+            if (rowIndex < 1)
+            {
+                throw new ArgumentOutOfRangeException("Row Size can't be Zero");
+            }
+            else
+            {
+                RowIndex = rowIndex;
+            }
+
+            if (colIndex < 1)
+            {
+                throw new ArgumentOutOfRangeException("Column Size can't be Zero");
+            }
+            else
+            {
+                ColumnIndex = colIndex;
+            }
+
+            Elements = new List<(int, int, int)>();
+        }
+
+
+        public int this[int row, int column]
+        {
+            get
+            {
+
+                var tempList = Elements.Where(x => x.row == row && x.column == column).Select(x => x.value).ToList();
+                if (tempList.Count < 1)
+                {
+                    return 0;
+                }
+                else
+                {
+                    return tempList[0];
+                }
+
+            }
+            set
+            {
+                Elements.Add((row, column, value));
```

Comment:
Consider sparse matrix sized 1mln x 1mln with single non-null element at [0,0] index.
It takes 4tb space and keeps single number.

Do not keep zeroes